compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -2842,6 +2856,15 @@ private STNode parseStatement(SyntaxKind tokenKind, STNode annots) {
                     finalKeyword = STNodeFactory.createEmptyNode();
                     return parseVariableDecl(getAnnotations(annots), finalKeyword, false);
                 }
+                //If the nex token is pipe token then it can be unary type or compound assignment statement.
+                if (nextToken.kind == SyntaxKind.PIPE_TOKEN) {
+                    STToken nextNextToken = peek(3);
+                    // if the nextNextToken is not an equal token then it is a unary type with user defined type
+                    if (nextNextToken.kind != SyntaxKind.EQUAL_TOKEN) {
+                        finalKeyword = STNodeFactory.createEmptyNode();
+                        return parseVariableDecl(getAnnotations(annots), finalKeyword, false);
+                    }
+                }
```

Comment:
I think we should remove from L#2851-L#2867. This should be handled inside the `parseStatementStartsWithIdentifier()` method. Similar cases are handled in that method.

Reply:
will look at it

Reply:
Changed. Please check that change is enough.